recipe/scripts/deactivate.sh

```diff
@@ -1,11 +1,22 @@
-export JAVA_HOME=$JAVA_HOME_CONDA_BACKUP
-unset JAVA_HOME_CONDA_BACKUP
-if [ -z $JAVA_HOME ]; then
-    unset JAVA_HOME
+# First check whether the backup variables are set.
+# The backed up variables are allowed to be set empty.
+# Then restore the backup, and unset the backup.
+
+# Note that the check whether the backup is set, is essential.
+# There are situations where conda executes the deactivate
+# script without having called the activate script. Without
+# this check, the deactivate script would unset JAVA_HOME in
+# those situations.
+
+# One such situation occurs when deactivating the environment
+# after installing openjdk for the first time.
```

Comment:
Thanks for that, I was struggling to figure out legitimate scenarios where deactivation is run without prior activation.